services/provider/server/server.go

```diff
@@ -338,7 +338,7 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 				keyProp = "adminKey"
 			}
 			extR = append(extR, &pb.ExternalResource{
-				Name: clientSecretName,
+				Name: strings.ReplaceAll(clientSecretName, "-"+consumerResource.Name, ""),
```

Comment:
Why are we shortening the names after the fact? 
I think it will be preferable to just create a shorter name in the first place. 
We own that code and can change it there.